back/src/domains/convention/use-cases/notifications/NotifyConventionReminder.ts

```diff
@@ -163,29 +167,33 @@ export class NotifyConventionReminder extends TransactionalUseCase<
       throw new ForbiddenError(
         forbiddenUnsupportedStatusMessage(conventionRead, reminderKind),
       );
-    await Promise.all(
-      [
-        ...agency.counsellorEmails.map(
-          (email) =>
-            ({
-              role: "counsellor",
-              email,
-            }) satisfies EmailWithRole,
-        ),
-        ...agency.validatorEmails.map(
-          (email) =>
-            ({
-              role: "validator",
-              email,
-            }) satisfies EmailWithRole,
-        ),
-      ].map((emailWithRole) =>
-        this.#sendAgencyReminderEmails(
-          emailWithRole,
-          conventionRead,
-          agency,
-          uow,
-          reminderKind,
+
+    await this.#saveNotificationsBatchAndRelatedEvent(
+      uow,
+      await Promise.all(
+        [
+          ...agency.counsellorEmails.map(
+            (email) =>
+              ({
+                role: "counsellor",
+                email,
+              }) satisfies EmailWithRole,
+          ),
+          ...agency.validatorEmails.map(
+            (email) =>
+              ({
+                role: "validator",
+                email,
+              }) satisfies EmailWithRole,
+          ),
+        ].map((emailWithRole) =>
+          this.#createAgencyReminderEmail(
+            emailWithRole,
+            conventionRead,
+            agency,
+            uow,
+            reminderKind,
+          ),
         ),
```

Comment:
c'est ici que ça change, on utilise l'utilitaire : `this.#saveNotificationsBatchAndRelatedEvent` pour que ça se fasse par batch